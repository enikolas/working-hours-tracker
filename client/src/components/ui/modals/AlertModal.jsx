import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../Modal';
import strings from '../../../utils/strings';

const AlertModal = ({
	active,
	title,
	content,
	onClose
}) => (
	<Modal
		active={active}
		title={title}
		content={content}
		buttons={[
			{ action: onClose, label: strings.ok }
		]}
	/>
);

AlertModal.propTypes = {
	active: PropTypes.bool,
	title: PropTypes.string,
	content: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.element
	]),
	onClose: PropTypes.func
};

AlertModal.defaultProps = {
	active: false,
	title: strings.warning,
	content: '',
	onClose: () => {}
};

export default AlertModal;
