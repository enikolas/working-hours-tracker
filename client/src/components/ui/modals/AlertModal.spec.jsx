import React from 'react';
import { shallow } from 'enzyme';

import AlertModal from './AlertModal';

describe('AlertModal', () => {
	it('should render properly without any props', () => {
		const wrapper = shallow(<AlertModal />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with a new title name', () => {
		const wrapper = shallow(<AlertModal title="this is a alert modal title" />);
		expect(wrapper).toMatchSnapshot();
	});
});
