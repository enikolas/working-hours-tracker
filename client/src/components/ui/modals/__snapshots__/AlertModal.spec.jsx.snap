// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal should render properly without any props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertModal
    active={false}
    content=""
    onClose={[Function]}
    title="Atenção"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "buttons": Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ],
      "cancel": null,
      "content": "",
      "hasButtons": true,
      "title": "Atenção",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "buttons": Array [
          Object {
            "action": [Function],
            "label": "OK",
          },
        ],
        "cancel": null,
        "content": "",
        "hasButtons": true,
        "title": "Atenção",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AlertModal should render with a new title name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertModal
    active={false}
    content=""
    onClose={[Function]}
    title="this is a alert modal title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "buttons": Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ],
      "cancel": null,
      "content": "",
      "hasButtons": true,
      "title": "this is a alert modal title",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "buttons": Array [
          Object {
            "action": [Function],
            "label": "OK",
          },
        ],
        "cancel": null,
        "content": "",
        "hasButtons": true,
        "title": "this is a alert modal title",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
